api/templates/model.ts.njk

```diff
@@ -24,14 +24,14 @@ import { fromNullable, Option } from "fp-ts/lib/Option";
 
 export interface {{ definitionName }} {
 {% for propName, prop in definition.properties -%}
```

Comment:
actually, the logic to handle object properties is non-recursive. this means two thing:

1. it only works for first level object properties (so you must create a dedicated object definition for every nested property)
2. you have to duplicate all the validation logic for object properties *and* for single properties

consider to handle all this logic using nunjucks macros: 
https://mozilla.github.io/nunjucks/templating.html#macro 
(one for every property type and a recursive macro for objects)